src/Database/DatabaseUtil.php

```diff
@@ -566,6 +567,10 @@ public function composeWhereForQueryBuilder(QueryBuilder $queryBuilder, string $
             $wildcard = str_replace('.', '_', $wildcard);
```

Comment:
WIldcard ist just the parameter name, if I read the code correcty? Wouldn't it be enought then, to check if a parameter name already exist and generate a new one by adding a suffix? So we would not need a manual additional parameter here.

Reply:
Wrong line referenced? Did you mean line 570...572?

Reply:
what kind of prefix should this be? some random string? if so then this PR would not change anything, bcuz i want to address a specific Wildcard to change its value, e.g. in a EventListener

Reply:
@Defcon0 Shure :D (but since my comment was more generic, it wasn't so important)

@AlexejKossmann Therefore PRs have a description field to explain, why you do what you do ;)

What comes in mind: is it possible, that we need to add more parameters for this method in the future? If the answear !== no, then we should think about changing the parameter type to array with an id index, otherwise the parameter list will become even longer

Reply:
@koertho is right. @AlexejKossmann Please add a parameter `array $options = []` and rename your `id` to `wildcardSuffix` and check it this way:

```
$wildcardSuffix = $options['wildcardSuffix'] ?? '';
```

OK for you both?

Reply:
@Defcon0 yes this is a nice solution! Adjusted the PR